src/main/java/com/bootcamp/demo/service/UserService.java

```diff
@@ -0,0 +1,151 @@
+package com.bootcamp.demo.service;
+
+import com.bootcamp.demo.model.User;
+import com.bootcamp.demo.service.exception.ItemNotFoundException;
+import com.bootcamp.demo.service.exception.ServiceException;
+import com.google.api.core.ApiFuture;
+import com.google.cloud.firestore.DocumentSnapshot;
+import com.google.cloud.firestore.Firestore;
+import com.google.cloud.firestore.QueryDocumentSnapshot;
+import com.google.cloud.firestore.WriteResult;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.ExecutionException;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+@Service
+public class UserService implements IUserService {
+    private static final Logger LOGGER = LoggerFactory.getLogger(UserService.class);
+    private static final String COLLECTION_USERS_PATH = "bookings/databases/users";
+    private final Firestore db;
+
+    public UserService(Firestore db) {
+        this.db = db;
+    }
+
+    @Override
+    public User createUser(final User user) {
+        LOGGER.info("CREATE USER - service function invoked: user = {}", user.toString());
+
+        try {
+            try {
```

Comment:
It seems a bit weird to have nested `try`s. Also, perhaps this could be simplified to sth like:
- find user by email
- if exists -> throw exception (or update if you think it would be suitable)
- if not exists -> create the user

Note that `set` can be used both for create & update.